src/Main.js

```diff
@@ -32,26 +48,38 @@ class Main extends React.Component {
   }
 
   render() {
-    return !this.state.isExtendedView ? (
-      <main className='main-section'>
-        <div className='all-movies-container'>
-          <Posters title='All Movies' movieData={this.state.movieData} changeExtendedState={this.changeExtendedState} />
-          <Posters title='More Movies' movieData={this.state.movieData} changeExtendedState={this.changeExtendedState} />
-          <Posters title='Even More Movies' movieData={this.state.movieData} changeExtendedState={this.changeExtendedState} />
-        </div>
-        <List movieData={this.state.movieData}/>
-      </main>
-    ) : (
-      <main className='main-section'>
-        <div className='single-movie-container'>
-          <ExtendedView
-            singleMovieData={this.state.movieData}
-            id={this.state.clickedPosterID}
-            changeExtendedState={this.changeExtendedState}
-            />
-        </div>
-      </main>
-    )
+    console.log("render")
+    if (this.state.loading) {
+      console.log("movie data", this.state.movieData)
+      console.log("loading status", this.state.loading)
+      return (
+        <p>'Loading...'</p>
+      )
+    }
+    if (!this.state.loading) {
+      console.log("movie data", this.state.movieData)
+      console.log("loading status", this.state.loading)
+      return !this.state.isExtendedView ? (
+        <main className='main-section'>
+          <div className='all-movies-container'>
+            <Posters title='All Movies' movieData={this.state.movieData} changeExtendedState={this.changeExtendedState} />
+            <Posters title='More Movies' movieData={this.state.movieData} changeExtendedState={this.changeExtendedState} />
+            <Posters title='Even More Movies' movieData={this.state.movieData} changeExtendedState={this.changeExtendedState} />
+          </div>
+          <List movieData={this.state.movieData}/>
+        </main>
+      ) : (
+        <main className='main-section'>
+          <div className='single-movie-container'>
+            <ExtendedView
+              singleMovieData={this.state.movieData}
+              id={this.state.clickedPosterID}
+              changeExtendedState={this.changeExtendedState}
+              />
+          </div>
+        </main>
+      )
+    }
```

Comment:
Conditional rendering for the win!

I think that this works really well. I wonder if there is a better/more efficient way we could refactor this?